Sources/ApolloCodegenLib/FileGenerators/OperationIdentifiersFileGenerator.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import ApolloUtils
+import OrderedCollections
+
+/// Representation of an operation that supports Automatic Persisted Queries
+struct OperationIdentifier {
+  let hash: String
+  let name: String
+  let source: String
+}
+
+/// File generator to create a JSON formatted file of operation identifiers.
+struct OperationIdentifiersFileGenerator {
+  /// File path of the generated target.
+  let filePath: String
+
+  /// `OperationIdentifier` objects to be written to `path`.
+  private var operationIdentifiers: [OperationIdentifier] = []
+
+  /// Designated initializer.
+  ///
+  /// Parameters:
+  ///  - config: A configuration object specifying output behavior.
+  init?(config: ApolloCodegen.ConfigurationContext) {
```

Comment:
This uses a failable initializer which I prefer since it places the logic of path evaluation in this struct. The alternative is to always initialize an instance but then `collectOperationIdentifier` would need to evaluate `path` for a quick exit; I think that's inefficient.

Reply:
Yeah, I'm down with the approach!